modules/seqexec/server/src/main/scala/seqexec/server/gems/GemsController.scala

```diff
@@ -3,34 +3,118 @@
 
 package seqexec.server.gems
 
-import seqexec.server.tcs.Gaos.{PauseCondition, ResumeCondition}
+import cats.Eq
+import cats.implicits._
+import seqexec.server.gems.Gems.GemsGuiderStatus
+import seqexec.server.tcs.Gaos.{PauseConditionSet, PauseResume, ResumeConditionSet}
 import squants.Time
 
 trait GemsController[F[_]] {
   import GemsController._
 
-  def pause(reasons: Set[PauseCondition]): F[Unit]
-  def resume(reasons: Set[ResumeCondition], config: GemsConfig): F[Unit]
+  def pauseResume(config: GemsConfig, pauseReasons: PauseConditionSet,
+                  resumeReasons: ResumeConditionSet): F[PauseResume[F]]
   def observe(expTime: Time): F[Unit]
   def endObserve: F[Unit]
 
+  val stateGetter: GemsGuiderStatus[F]
+
 }
 
 object GemsController {
 
-  sealed trait GemsConfig
+  sealed trait P1Usage
+  object P1Usage {
+    case object UsesP1 extends P1Usage
+    case object DoesNotUseP1 extends P1Usage
+
```

Comment:
Since you need it elsewhere when parsing `Boolean`s from the sequence(?), it's probably best to define `fromBoolean` here:

```scala
def fromBoolean(b: Boolean): P1Usage =
  if (b) UsesP1 else DoesNotUseP1
```